python/feathub/processors/flink/table_builder/metric_store_utils.py

```diff
@@ -0,0 +1,151 @@
+#  Copyright 2022 The FeatHub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import collections
+import json
+import typing
+
+from feathub.common.exceptions import FeathubException
+from feathub.feature_tables.sinks.prometheus_sink import PrometheusSink
+from feathub.feature_tables.sinks.sink import Sink
+from feathub.feature_views.derived_feature_view import DerivedFeatureView
+from feathub.feature_views.feature import Feature
+from feathub.feature_views.sliding_feature_view import SlidingFeatureView
+from feathub.feature_views.transforms.sliding_window_transform import (
+    SlidingWindowTransform,
+)
+from feathub.metric_stores.metric import Metric, Count, Ratio
+from feathub.metric_stores.metric_store import MetricStore
+from feathub.metric_stores.prometheus_metric_store import PrometheusMetricStore
+from feathub.processors.flink.table_builder.source_sink_utils_common import (
+    generate_random_name,
+)
+from feathub.processors.materialization_descriptor import MaterializationDescriptor
+from feathub.table.table_descriptor import TableDescriptor
+
+
+def _get_metric_name(metric: Metric, metric_store: MetricStore) -> str:
+    return f"feathub_{metric_store.namespace}_feature_{metric.metric_type}"
+
+
+def _get_metric_tags(metric: Metric, feature: Feature) -> typing.OrderedDict[str, str]:
+    tags = collections.OrderedDict(
+        [
+            ("feature_name", feature.name),
+            ("window_time", str(metric.window_size)),
+        ]
+    )
+    if isinstance(metric, Count) or isinstance(metric, Ratio):
```

Comment:
It seems more extensible to move the logic of determining the metric tags to the metric class rather than enumerating all metric subclasses in a util class here.